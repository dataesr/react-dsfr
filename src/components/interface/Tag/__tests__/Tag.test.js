import { render, screen } from '@testing-library/react';
import { Tag, TagGroup } from '..';

describe('<Tag />', () => {
  it('should renders Tag properly', () => {
    render(<Tag as="a" href="/" title="titre">tag-test</Tag>);
    const tag = screen.getByText('tag-test');
    expect(tag).toBeInTheDocument();
    expect(tag).toMatchSnapshot();
  });
  it('should renders small Tag properly', () => {
    render(<Tag size="sm" as="span">tag-test</Tag>);
    const tag = screen.getByText(/tag-test/i);
    expect(tag).toBeInTheDocument();
    expect(tag.className).toBe('fr-tag fr-tag--sm');
    expect(tag).toMatchSnapshot();
  });
  it('should renders TagGroup properly', () => {
    render(
      <TagGroup>
        <Tag size="sm" as="span">tag-test</Tag>
        <Tag size="sm" as="span">tag-test</Tag>
        <Tag size="sm" as="span">tag-test</Tag>
        <Tag size="sm" as="span">tag-test</Tag>
      </TagGroup>,
    );
    const tagGroup = screen.getByTestId('tag-group');
    expect(tagGroup).toBeInTheDocument();
    const tags = screen.getAllByText('tag-test');
    expect(tags[0].className).toBe('fr-tag fr-tag--sm');
    expect(tags).toHaveLength(4);
    expect(tagGroup).toMatchSnapshot();
  });
});
