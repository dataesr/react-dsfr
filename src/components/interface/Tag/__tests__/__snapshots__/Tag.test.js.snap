// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should renders Tag properly 1`] = `
<a
  class="fr-tag fr-tag--md"
  href="/"
  title="titre"
>
  tag-test
</a>
`;

exports[`<Tag /> should renders TagGroup properly 1`] = `
<ul
  class="fr-tags-group"
  data-testid="tag-group"
>
  <li>
    <span
      class="fr-tag fr-tag--sm"
    >
      tag-test
    </span>
  </li>
  <li>
    <span
      class="fr-tag fr-tag--sm"
    >
      tag-test
    </span>
  </li>
  <li>
    <span
      class="fr-tag fr-tag--sm"
    >
      tag-test
    </span>
  </li>
  <li>
    <span
      class="fr-tag fr-tag--sm"
    >
      tag-test
    </span>
  </li>
</ul>
`;

exports[`<Tag /> should renders small Tag properly 1`] = `
<span
  class="fr-tag fr-tag--sm"
>
  tag-test
</span>
`;
